### PR TITLE
PLANNER-737: Workbench: scoreHolder global variable generation based on the PlanningSolution score type

### PR DESCRIPTION
**Pull request by mcimbora**
Based on https://github.com/droolsjbpm/kie-wb-common/pull/693 (not integrated yet), changes included in https://github.com/droolsjbpm/kie-wb-common/commit/0c6301c2332b070a94b2d09c962e3338fc937b6d

Add Save and Rename Helpers for Data Modeler. DataModelerRenameWorkaroundHelper is a temporary solution until we decide to switch to using RenameService when performing rename operation in DataModelerServiceImpl (currently there's a workaround which uses IOService.move).

Part of:
https://github.com/droolsjbpm/kie-wb-common/pull/699
https://github.com/droolsjbpm/optaplanner-wb/pull/136

### CODE REVIEW

File: kie-wb-common-screens/kie-wb-common-data-modeller/kie-wb-common-data-modeller-backend/src/main/java/org/kie/workbench/common/screens/datamodeller/backend/server/DataModelerServiceImpl.java [hunk 1]
```diff
@@ -758,6 +772,12 @@ public Path renameWorkaround( final Path path,
                                 serviceHelper.makeCommentedOption( "File [" + path.toURI() + "] renamed to [" + targetPath.toURI() + "]." )
                               );
 
+                if ( renameHelperInstance != null ) {
+                    for ( DataModelerRenameWorkaroundHelper renameHelper : renameHelperInstance ) {
+                        renameHelper.postProcess( Paths.convert( _path ), Paths.convert( _target ) );
```
Comment (manstis) [inline kie-wb-common-screens/kie-wb-common-data-modeller/kie-wb-common-data-modeller-backend/src/main/java/org/kie/workbench/common/screens/datamodeller/backend/server/DataModelerServiceImpl.java]: We can hope the Java compiler moves these conversions outside the loop....

File: kie-wb-common-screens/kie-wb-common-data-modeller/kie-wb-common-data-modeller-backend/src/main/java/org/kie/workbench/common/screens/datamodeller/backend/server/DataModelerServiceImpl.java [hunk 2]
```diff
@@ -570,7 +578,12 @@ public GenerationResult saveSource( final String source,
                                  metadataService.setUpAttributes( path, metadata ),
                                  serviceHelper.makeCommentedOption( commitMessage ) );
                 result.setPath( path );
+            }
 
+            if ( saveHelperInstance != null ) {
```
Comment (manstis) [inline kie-wb-common-screens/kie-wb-common-data-modeller/kie-wb-common-data-modeller-backend/src/main/java/org/kie/workbench/common/screens/datamodeller/backend/server/DataModelerServiceImpl.java]: Now you have possible additional ```IOService``` operations; should the whole ```saveSource(...)``` method use a VFS "batch" (at the moment it's discretionary *only* if the package has changed).

Comment (mcimbora) [inline kie-wb-common-screens/kie-wb-common-data-modeller/kie-wb-common-data-modeller-backend/src/main/java/org/kie/workbench/common/screens/datamodeller/backend/server/DataModelerServiceImpl.java]: Correct, I checked IOService implementation we use and the concern of RenameService invoking startBatch once more is groundless. Invocations performed within the same thread share the same lock instance and the RenameService decreases the lock counter in a finally block.

**Comment by mcimbora** [comment]
@manstis Updated, thanks for a review.

**Comment by manstis** [comment]
@mcimbora You decided against adding unit tests?

**Comment by mcimbora** [comment]
@manstis I just forget how I love writing unit tests :-) I added a couple of them for save/rename. Additionally I introduced better division of weld/mockito tests.

**Comment by mcimbora** [comment]
@manstis Cool. I'll feel even better if the tests actually pass, looking.

**Comment by mcimbora** [comment]
Jenkins retest this.

**Comment by mcimbora** [comment]
@manstis All green, I think this is good to go in.

**Pull request merged**